R/Lithuania.R

```diff
@@ -0,0 +1,407 @@
+#'    intermediate cause of death or other important pathological condition,
+#'    or identified as related to COVID-19 disease (coronavirus infection).
+#'    Deaths due to external causes are not included (ICD disease codes
+#'    are V00-Y36, or Y85-Y87, or Y89, or S00-T79, or T89-T98).
+#' 3. *`after`* Number of deaths from any cause of COVID-19 or COVID-19
+#'    deaths due to non-external causes within 28 days.
+#'    The indicator is calculated by summing all registered records of the
+#'    medical form E106 (unique persons), in which the ICD disease codes
+#'    U07.1, U07.2, U07.3, U07.4, U07 are indicated as the main, direct,
+#'    intermediate cause of death or other important pathological condition,
+#'    or identified as related to COVID-19 disease (coronavirus infection)
+#'    and all records of medical form E106 (unique individuals) where the
+#'    person died within the last 28 days after receiving a positive
+#'    diagnostic response to the SARS-CoV-2 test or had an entry in medical
+#'    form E025 with ICD disease code U07.2 or U07.1. Deaths due to external
+#'    causes are not included (ICD disease codes are V00-Y36, or Y85-Y87, or
+#'    Y89, or S00-T79, or T89-T98).
+#'
+#'  The number of deaths reported in the last day is preliminary and
+#'  increases by about 20-40% in a few days. Such a "delay" in the data is
+#'  natural: for example, for those who died last night, a death certificate
+#'  is likely to be issued as soon as this report is published this morning.
+#'
+#' @section De jure and statistical counts:
+#'
+#' Beginning in February 2021 the OSP makes statistical estimates
+#' of the number of recovered and active cases, since review of the data
+#' showed that some cases individuals still considered as active cases
+#' had recovered, but not documented or registered as such.
+#'
+#' These are listed as by the OSP as `active_de_jure` and
+#' `recovered_de_jure` (officially still considered sick),
+#' and `active_sttstcl` and `recovered_sttstcl` (an estimate of how
+#' many of these are still ill).
+#'
+# nolint start
+#' @source https://opendata.arcgis.com/datasets/d49a63c934be4f65a93b6273785a8449_0
+# nolint end
+#' @examples
+#' \dontrun{
+#' region <- Lithuania$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' }
+#' @export
+Lithuania <- R6::R6Class("Lithuania",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes
+    #' @field country name of country to fetch data for
+    country = "Lithuania",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1", "2"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list(
+      "1" = "county", # apskritis
+      "2" = "municipality" # savivaldybe
+    ),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list(
+      "1" = "iso_3166_2", # default is to just give county data
```

Comment:
Agree this is a nicer way of doing it and probably more in line with what people will expect.